src/PostgREST/AppState.hs

```diff
@@ -261,8 +261,10 @@ usePool AppState{stateObserver=observer, stateMainThreadId=mainThreadId, ..} ses
         SQL.ServerError{} ->
           when (Error.status (Error.PgError False err) >= HTTP.status500) $
             observer $ QueryErrorCodeHighObs err
-    SQL.SessionUsageError (SQL.QueryError _ _ (SQL.ClientError _)) ->
-      pure ()
+    err@(SQL.SessionUsageError (SQL.QueryError _ _ (SQL.ClientError _))) ->
+      -- logging since https://github.com/PostgREST/postgrest/issues/3841
+      when (Error.status (Error.PgError False err) >= HTTP.status500) $
```

Comment:
Hmm, `code-cov` is complaining on `False`. This is because we always return `503` here without evaluating `False`.

https://github.com/PostgREST/postgrest/blob/3f78615dff51f680b141c3063a328487efe29e29/src/PostgREST/Error.hs#L468

Reply:
> This is because we always return `503`.

Oh, right, since we always return `503` we don't need the `when...` condition then, only the observer, since this should always be logged.

As a side note, I'd remove the comment `-- logging since...`, or explain directly that [this usually indicates problems with connection](https://hackage.haskell.org/package/hasql-1.9.1.1/docs/Hasql-Session.html#v:ClientError) and returns `503`, so we log it.